### PR TITLE
Support multiple shells

### PR DESCRIPTION
**Pull request by wm**
I wanted to use this for my zsh shell so I updated it to allow multiple shells.

I also extracted most segment functionality into functions with an eye toward making this customizable.

Finally I updated the README with usage instructions.

### CODE REVIEW

File: powerline-shell.go [hunk 1]
```diff
@@ -25,6 +24,8 @@ import (
 	"path/filepath"
 	"regexp"
 	"strings"
+
+	"github.com/wm/powerline-shell-go/powerline"
```
Comment (sivel) [inline powerline-shell.go:30]: This should reference the parent repo and not your fork.

Comment (wm) [inline powerline-shell.go:30]: Addressed in e00204c

**Comment by sivel** [comment]
There seems to be a change in output between master and this PR.  The initial path tested was is `~/go/src/github.com/sivel/powerline-shell-go`

Before:
screenshot 2015-01-05 at 4 23 07 pm

After:
screenshot 2015-01-05 at 4 22 32 pm

You will notice the omission of the initial `go` from the path.

Also with `/private/tmp`:

Before:
screenshot 2015-01-05 at 4 28 05 pm

After:
screenshot 2015-01-05 at 4 27 47 pm

You will notice the addition of `/` at the front that shouldn't be there.  (Note that there may have been an initial bug for this last test, as it should have not had the `...` segment there.

**Comment by wm** [comment]
I fixed the cwd segments (in ca3f519) to not output the / unless you are in root. I fixed the bug you mentioned you had too (private > ... > tmp). 

I added some tests for addCwd since it is hard to understand what is happening when. I think the tests are straightforward though.

thanks for the feedback

**Comment by sivel** [comment]
Great, I'll look into testing the changes soon.

**Comment by sivel** [comment]
Looking better.  When a python virtual environment is present/activated, the venv should appear at the start.

This image shows the output from your PR first, and the original second.

screenshot 2015-01-06 at 11 35 20 am

**Comment by wm** [comment]
okay, I moved the virtual machine to the beginning.

**Comment by sivel** [comment]
This was manually merged a long time ago.  Not sure why I never closed the PR.

**Pull request closed without merging**